src/request.rs

```diff
@@ -1,119 +1,24 @@
-#[cfg(feature = "http")]
-extern crate reqwest;
-
-use error::{Error, RequestErrorKind};
-use parser::parse_response;
-use transport::Transport;
-use utils::escape_xml;
-use Value;
-
-use std::collections::BTreeMap;
 use std::io::{self, Write};
+use xml::escape::{escape_str_pcdata as escape_xml};
+
+use crate::Value;
 
 /// A request to call a procedure.
 #[derive(Clone, Debug)]
 pub struct Request<'a> {
     name: &'a str,
-    args: Vec<Value>,
+    args: &'a [Value],
 }
```

Comment:
It seems odd to me generally that this struct only has references--why not just copy the data into it and have:

```rust
pub struct Request {
    name: String,
    args: Vec<Value>,
}
```

I do support standardizing on one and not having both, though :)